pkg/util/purge/resourcegroups.go

```diff
@@ -32,14 +32,20 @@ func (rc *ResourceCleaner) CleanResourceGroups(ctx context.Context) error {
 }
 
 // cleanResourceGroup checkes whether the resource group can be deleted if yes proceed to clean the group in an order:
+//     - delete any ARO clusters
 //     - unassign subnets
 //     - clean private links
 //     - checks ARO presence -> store app object ID for futher use
 //     - deletes resource group
 func (rc *ResourceCleaner) cleanResourceGroup(ctx context.Context, resourceGroup mgmtfeatures.ResourceGroup) error {
-	if rc.shouldDelete(resourceGroup, rc.log) {
+	if rc.shouldDelete(resourceGroup, rc.log) && !rc.dryRun {
```

Comment:
I changed where the `dryRun` check is happening as I think that we shouldn't be attempting to update or clean up any resources (e.g. subnets, PLS, etc.) if we're in a dry run.  If there's a reason why we check for dry run later let me know.